cowprotocol/services
# Solver engine s3 upload

# Description
For debugging purposes it is useful to have access to historic auction instances in the legacy format (pre solver/driver co location) as a bunch of internal checks (EBBO and the like) still use the old format.

While we should change these scripts to work on the new solver engine dto, we can - for the time being - continue to upload the legacy instance files to S3 like we did pre co-location.

cc @fhenneke @harisang 

# Changes

- [x] Create Persistence infra in solvers crate
- [x] Allow legacy solver to specify S3 upload config
- [x] If set, upload legacy instance to S3 

## How to test
Ran a shadow setup locally, observe files getting uploaded to https://staging-bucket-133876976458.s3.eu-central-1.amazonaws.com/legacy/8254181.json

## Related Issues

Fixes #1752
